metrics/sinks/stackdriver/stackdriver.go

```diff
@@ -197,7 +202,12 @@ func (sink *StackdriverSink) ExportData(dataBatch *core.DataBatch) {
 		requests = append(requests, req)
 	}
 
-	go sink.sendRequests(requests)
+	select {
+	case sink.batchesQueue <- requests:
+		// queue not full, batch will be processed
+	default:
+		// queue full, dropping the batch
```

Comment:
Should we log a warning? Dropping batches is not a intended state

Reply:
+1 to a warning. Maybe also log the batch size and record a metric for number of dropped batches/metrics?

Reply:
Good point, added a warning.